src/comet.erl

```diff
@@ -0,0 +1,97 @@
+-module(comet).
+-behaviour(gen_server).
+
+-export([start_link/0]).
+
+%% API:
+-export([ start/0
+        , start/1
+        , stop/0
+        , insert/3
+        , batch_insert/3
+        , fetch_one/2
+        , fetch/3
+        , delete/3
+        , range_delete/4
+        ]).
+
+%% gen_server callbacks:
+-export([ init/1
+        , handle_call/3
+        , handle_cast/2
+        , handle_info/2
+        , terminate/2
+        ]).
+
+-define(SERVER, comet).
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% API functions
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+start() ->
+    start(<<>>).
+
+start(ClusterFile) -> 
+    gen_server:start_link(?MODULE, [ClusterFile], []).
+
+stop() -> 
+    gen_server:call(?MODULE, stop).
+
+start_link() -> 
+    gen_server:start_link(?MODULE, [], []).
+
+% FoundationDB can only be intiailized once
+% in a given OS process.
+-spec init([comet_types:cluster_file()]) -> {ok, any()}.
+init([ClusterFile]) -> 
+    process_flag(trap_exit, true),
+    Db = erlfdb:open(ClusterFile),
+    {ok, Db}.
+
+-spec insert(pid(), comet_types:comet_queue_name(), comet_types:comet_kv()) -> ok.
+insert(Pid, QueueName, Value) ->    
+    gen_server:call(Pid, {insert, QueueName, Value}).
+
+-spec batch_insert(pid(), comet_types:comet_queue_name(), list(comet_types:comet_kv())) -> ok.
```

Comment:
Nit: it would be great to create a new type for the FDB server's pid, something like `-type connection() :: pid()`. It won't affect the typechicking, but it will help documenting the code. The users will see that it's not just any pid, but specifically a FDB process pid.

Reply:
You are right!

Reply:
done